substrate-archive/Cargo.toml

```diff
@@ -0,0 +1,82 @@
+[package]
+name = "substrate-archive"
+version = "0.5.0-pre"
+authors = ["Andrew Plaza <andrew.plaza@parity.io>"]
+edition = "2018"
+
+[dependencies]
+# external
+log = "0.4"
+futures = { version = "0.3" }
+fdlimit = "0.2"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+serde-aux = "0.6.1"
+dirs = { version = "3", package = "directories" }
+sqlx = { version = "0.4.0-beta.1", default-features = false, features = ["postgres", "macros", "runtime-async-std", "migrate", "json", "offline"] }
+async-trait = "0.1"
+hex = "0.4"
+itertools = "0.9"
+# 0.7.2 introduces breaking changes with Send/Sync bounds
```

Comment:
Does this mean that `0.7.2` mustn't be used? Best use `flume = "=0.7.1` then, or something similar that stops `0.7.2` to be used.
If upgrading requires a refactor we don't want to do in this PR it would be good to add an issue for it.

Reply:
that's not introduced in this PR but `flume = "=0.7.1` seems like a greate idea without having any clue about the context here.

/cc @insipx I suppose you can explain this :)

Reply:
yeah, flume =0.7.1 should be the way to go

When flume first updated with 0.7.2 (despite the minor release), there was a breaking change in the function of their asynchronous `send_async` and `recv_async` methods because of the use of a `cell` in one of the Sinks. This resulted in not being able to use recv_async across futures bounds. Flume is on 0.9 now, however, and I believe that the issue has been resolved. 

This update should just be done in another PR though, it touches basically all of the actors and some more